kivy/uix/behaviors/hover.py

```diff
@@ -0,0 +1,284 @@
+            on_release: root.min_hov = self.hover_height
+
+        HovButton:
+            size_hint_x: 0.5
+            hover_height: 0
+            text: 'Set min_hover_height: 0'
+            on_release: root.min_hov = self.hover_height
+
+    Label:
+        size_hint: 0.9, None
+        height: self.texture_size[1]
+        pos: root.width * 0.05, sizebox.top + (self.height * 0.5)
+        text_size: self.size
+        text: "Buttons that change HoverBehavior min_hover_height to their own height, useful when you want to disable hovering for lower widgets after opening a popup or putting something else on top"
+
+    HovButton:
+        id: tggrab
+        size_hint: 1, 0.12
+        pos: 0, 0
+        text: 'Toggle grab hover'
+        on_release: setattr(root, "hov_grab", None) if root.hov_grab else setattr(root, "hov_grab", self)
+""")
+
+runTouchApp(RootWidget())
+
+
+See :class:`~kivy.uix.behaviors.HoverBehavior` for details.
+'''
+
+__all__ = ('HoverBehavior', )
+
+from kivy.properties import BooleanProperty, NumericProperty, ObjectProperty
+from kivy.weakproxy import WeakProxy
+from kivy.core.window import Window
+
+
+class HoverBehavior(object):
+    '''
+    The HoverBehavior `mixin <https://en.wikipedia.org/wiki/Mixin>`_ provides
+    Hover behavior. When combined with a widget, hovering mouse cursor
+    above it's position will call it's on_hovering event. Please see
+    the :mod:`hover behaviors module <kivy.uix.behaviors.hover>` documentation
+    for more information.
+
+    .. versionadded:: 1.10.1
+    '''
+
+    _hover_grab_widget = None
+    '''WeakProxy of widget which has grabbed focus or None'''
+
+    _hover_widgets = []
+    '''List of hover widget WeakProxy references'''
+
+    min_hover_height = 0
+    '''Numeric class attribute of minimum height where "hovering" property
+    will be updated'''
+
+    hovering = BooleanProperty(False)
+    '''Hover state, is True when mouse enters it's position
+
+    :attr:`hovering` is a :class:`~kivy.properties.BooleanProperty` and
+    defaults to `False`.
+    '''
+
+    hover_height = NumericProperty(0)
+    '''Number that is compared to min_hover_height.
+
+    :attr:`hover_height` is a :class:`~kivy.properties.NumericProperty` and
+    defaults to `0`.
+    '''
+
+    def _on_hover_mouse_move(win, pos):
+        '''Internal method that is binded on Window.mouse_pos.
+        Compares mouse position with widget positions,
+        ignoring disabled widgets and widgets with hover_height below
+        HoverBehavior.min_hover_height,
+        then sets widget hovering property to False or True'''
+
+        collided = [] # widget weak proxies that collide with mouse pos
+
+        if HoverBehavior._hover_grab_widget:
+            HoverBehavior.hover_widget_refs = [HoverBehavior._hover_grab_widget]
+        else:
+            HoverBehavior.hover_widget_refs = HoverBehavior._hover_widgets
+        for ref in HoverBehavior.hover_widget_refs:
```

Comment:
Personally, I think there's too much stuff going on that's bound to **`mouse_pos`** which **isn't** a stable number because:

* not an integer (I have float tuple, but it's from SDL2, so...)
* when a user uses touchpad the input touch isn't sane (fat fingers for example, bad touch surface, etc)

therefore if you bind anything to that, it has to be flat. Such thing like **nested** for loops is quite bad unless the content you iterate over is really really small(or no 60fps). Now imagine someone to add that on each widget. Freeze at best :D